Source/SwiftLintFramework/Protocols/Rule.swift

```diff
@@ -32,11 +37,32 @@ public protocol ConfigurationProviderRule: Rule {
 }
 
 public protocol CorrectableRule: Rule {
+    func correct(file: File, compilerArguments: [String]) -> [Correction]
     func correct(file: File) -> [Correction]
 }
 
+public extension CorrectableRule {
+    func correct(file: File, compilerArguments: [String]) -> [Correction] {
+        return correct(file: file)
+    }
+}
+
 public protocol SourceKitFreeRule: Rule {}
 
+public protocol CompilerArgumentRule: Rule {}
+
+public extension CompilerArgumentRule {
+    func validate(file: File) -> [StyleViolation] {
```

Comment:
These default extensions don't seem like they would ever work for this type of rule? Are they required or you could omit them and require implementors to add them?